docs/advanced/storage.md

```diff
@@ -0,0 +1,51 @@
+# Storage
```

Comment:
I think it makes the most sense to have it as an overarching doc for all of daffodil rather than have it reside in core. People might not find it easily if it is in core.

Reply:
This doc is about storage, which is a part of the API of Daffodil core. I would expect that docs provided at this level are more like "integration" guides, which describe tutorials on how to actually use the Persistence APIs, not simply the list of available storage APIs and their purposes.

Reply:
I agree with Damien, most of this guide is specific to core. maybe `Environment-Specific Storage Services` should be in the main docs folder but everything else belongs in core. Moved to core.

Reply:
I still disagree, because I think this is more of an "integration" guide that happens to have api docs in it rather than api docs that happen to have tutorials. I agree that the API docs should probably go in core as regular, autogenerated docs that would be handled by dgeni. I disagree that this is essentially an api doc; I think it is a tutorial guide on how to use the persistence APIs.

Reply:
I'm fine with resolving this though. No need for this to hold up the PR, I don't think.

Reply:
@damienwebdev bump. This guide has been moved to core